src/java.base/share/native/libjli/java.c

```diff
@@ -1626,7 +1626,7 @@ TranslateApplicationArgs(int jargc, const char **jargv, int *pargc, char ***parg
     for (i = 0; i < jargc; i++) {
         const char *arg = jargv[i];
         if (arg[0] == '-' && arg[1] == 'J') {
-            *nargv++ = ((arg + 2) == NULL) ? NULL : JLI_StringDup(arg + 2);
+            *nargv++ = (arg[2] == '\0') ? NULL : JLI_StringDup(arg + 2);
```

Comment:
Wow!

Reply:
I wonder if the client expects NULL strings in the result, or if the NULL value should be an empty string?  If empty strings are okay, this would be simpler without the conditional, just dup from arg + 2 to the terminating byte (which might be immediate).

Reply:
`NULL` affects as a loop stopper in `ParseArguments()` as following:

```c
static jboolean
ParseArguments(int *pargc, char ***pargv,
               int *pmode, char **pwhat,
               int *pret, const char *jrepath)
{
    int argc = *pargc;
    char **argv = *pargv;
    int mode = LM_UNKNOWN;
    char *arg;

    *pret = 0;

    while ((arg = *argv) != 0 && *arg == '-') {
```

But I'm not sure it is valid, I think it might be discussed as another issue.